### DISCUSSION TITLE
413 request entity too large on Route handler

### DISCUSSION THREAD

**Discussion by ricardomatias** [vercel/next.js#70621] [App Router]
What is the magic incantation that I have to add so that I can upload large images via a form?

**Comment by Juneezee** [comment]
Unfortunately, it is not possible to set body size limit for Route Handlers in App Router (https://github.com/vercel/next.js/issues/57501).

You might want to use Server Action and set the `bodySizeLimit`, see https://nextjs.org/docs/app/api-reference/next-config-js/serverActions#bodysizelimit

**Comment by icyJoseph** [comment]
Hi,

Often times, when it comes to images, and other kind of asset uploads, it is better to use a dedicated service, for example Cloudinary, where you upload the image to, and get back a URI, which is then used together with the form submission, so that your DB can save the location of such asset. S3 buckets, and alike technology, are also an alternative, etc...

Here's a bit of reasoning from the docs, regarding the size limits, in this case for Pages Router, but it is the same reason at large, for App Router:

> The "API Routes Response Size Limited to 4MB" error arises when your API Route is trying to send a response larger than the allowed size of 4MB. API Routes in Next.js are designed to deliver quick responses and are not built to support the transmission of large amounts of data.

And the implicit recommendation to use a CDN:

> If you are not using Next.js in a serverless environment, and understand the performance implications of not using a CDN or dedicated media host
